src/Adapter/Product/AbstractProductHandler.php

```diff
@@ -61,6 +62,8 @@ protected function getProduct(ProductId $productId): Product
                     $productIdValue
                 ));
             }
+
+            $this->setUnitPrice($product);
```

Comment:
I need to set unit_price manually to be able to use Number for calculations. And the unit_price is not calculated when simply creating Product object (it requires passing additional argument to load $full object and possibly something more)